library/RemoteServer.cpp

```diff
@@ -476,11 +476,14 @@ void ServerMainImpl::threadFn(std::promise<bool> promise, int port)
     CActiveSocket *client = nullptr;
 
     try {
-        while ((client = server.socket.Accept()) != NULL)
+        while (true)
         {
-            BlockGuard lock;
-            ServerConnection::Accepted(client);
-            client = nullptr;
+            if ((client = server.socket.Accept()) != NULL)
```

Comment:
my main concern here that you are not guaranteed that `socket` will be in a valid state after `Accept` fails. if the socket is an invalid state, all subsequent calls to `Accept` will fail, which will lead to livelock

you need to verify that the socket is still in a state able to accept new connections after a failure before making another attempt; if you can't do that you have to close that listener and create a new one

Reply:
Ouch, committed an corrected version that checks if the socket is valid before attempting to accept, not entirely sure that is enough tho. 

Added an error output as well so I can diagnose my actual problem better.

Reply:
This might not help, since all `isSocketValid` does is check if the socket descriptor is not -1; it does not check to see if the underlying socket is still able to accept connections as that would require asking the OS if the socket is still in a valid state, which `clsocket` is not programmed to do. I checked the POSIX documentation (from Open Group), and if there a reliable way to interrogate a listener as to whether it is still able to accept connections, I can't find it, nor do I remember there being one from the days I actually did low-level networking programming in UNIX.

So what I'd probably do here is have a failure counter, which gets reset to zero after each successful accept, but gets incremented after each failed accept. If the counter gets too high (3 or 5 or something), close the listening socket and attempt to allocate a new one, exiting if that fails (again, possibly trying more than once).

The sorts of circumstances in which `accept` is likely fail persistently are fairly dire (out of memory, out of kernel resource, the interface on which the socket was allocated has been shut down) but good code should still deal with such circumstances as gracefully as possible.

Reply:
Will make an attempt at it, the occasions accept would fail shouldn't be to often so that the listening socket is disconnected shouldn't be to big of a problem.

Thanks for the feedback!